src/maud/running_stan.py

```diff
@@ -116,6 +117,35 @@ def simulate(mi: MaudInput, output_dir: str, n: int) -> CmdStanMCMC:
     )
 
 
+def optimize(mi: MaudInput, output_dir: str) -> CmdStanMLE:
+    """Generate MLE from the input maud model.
+
+    :param mi: a MaudInput object
+    :param output_dir: a string specifying where to save the output.
+    """
+    model = cmdstanpy.CmdStanModel(
+        stan_file=os.path.join(HERE, STAN_PROGRAM_RELATIVE_PATH),
+        cpp_options=mi.config.cpp_options,
+        stanc_options=mi.config.stanc_options,
+    )
+    set_up_output_dir(output_dir, mi)
+    return model.optimize(
+        output_dir=output_dir,
+        # upper bound, will stop once the gradients become small enough
+        iter=3000,
+        data=os.path.join(output_dir, "input_data_train.json"),
+        inits=os.path.join(output_dir, "inits.json"),
+        algorithm="LBFGS",
+        init_alpha=1e-4,
+        tol_param=1e-12,
+        history_size=10,
+        show_console=True,
+        refresh=1,
+        save_profile=True,
+        save_iterations=True,
+    )
```

Comment:
Could the arguments here be made customisable as for `variational` and `sample`?

Reply:
Done!